flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperator.java

```diff
@@ -323,7 +323,9 @@ protected boolean isUsingCustomRawKeyedState() {
     public void open() throws Exception {}
 
     @Override
-    public void finish() throws Exception {}
+    public void finish() throws Exception {
+        setCurrentKey(EndOfDataKey.INSTANCE);
```

Comment:
Why don't you use `null`? This must not work as the `EndOfDataKey` is not of the same type as user's key.

Reply:
Hi @dawidwys ,
Actually if we set null as the key and run in streaming mode, we will get:
`
Caused by: java.lang.NullPointerException: Assigned key must not be null!
	at org.apache.flink.util.Preconditions.checkNotNull(Preconditions.java:76)
	at org.apache.flink.runtime.state.KeyGroupRangeAssignment.assignToKeyGroup(KeyGroupRangeAssignment.java:64)
	at org.apache.flink.runtime.state.AbstractKeyedStateBackend.setCurrentKey(AbstractKeyedStateBackend.java:191)
	at org.apache.flink.streaming.api.operators.StreamOperatorStateHandler.setCurrentKey(StreamOperatorStateHandler.java:362)
	... 22 more
`

Reply:
I see. We need to come up with a different approach then, unfortunately. We can not simply choose a random key. :(

Let me think about it for a bit, and get back to you. Unless, you have a different idea.